### PR TITLE
Spark: Update Hadoop/AWS Sdk version + set all to Spark3

### PR DESCRIPTION
**Pull request by akhurana001**
# TL;DR

- Updates Hadoop/AWS version to support IAM role fetch via https://docs.aws.amazon.com/eks/latest/userguide/iam-roles-for-service-accounts-minimum-sdk.html
-  Default flytekit[all] to Spark 3.x (https://github.com/lyft/flyte/issues/483)

## Type
 - [ ] Bug Fix
 - [ ] Feature
 - [ ] Plugin

## Are all requirements met?

 - [ ] Code completed
 - [ ] Smoke tested
 - [ ] Unit tests added
 - [ ] Code documentation added
 - [ ] Any pending items have an associated Issue

## Complete description
 _How did you fix the bug, make the feature etc. Link to any design docs etc_

## Tracking Issue
https://github.com/lyft/flyte/issues/<number>

## Follow-up issue
_NA_
OR
_https://github.com/lyft/flyte/issues/<number>_

### CODE REVIEW

**Review by wild-endeavor — COMMENTED** [review]
Can you change this to be a PR into the `annotations` branch?

Also, on that branch, this information is currently in two places  - the lazy loader file, and also the main `setup.py` file.  It'll be merged back shortly but could you just change it in both places for now?

**Comment by akhurana001** [comment]
> Can you change this to be a PR into the `annotations` branch?
> 
> Also, on that branch, this information is currently in two places - the lazy loader file, and also the main `setup.py` file. It'll be merged back shortly but could you just change it in both places for now?

Created https://github.com/lyft/flytekit/pull/334 for `annotations` branch

**Comment by akhurana001** [comment]
closing in favor of https://github.com/lyft/flytekit/pull/334/files
